R/forecast.R

```diff
@@ -253,29 +265,45 @@ validate_forecast.forecast_point <- function(forecast, forecast_type = NULL, ...
     )
     #nolint end
   }
-  return(forecast[])
+  return(invisible(NULL))
 }
 
 
 #' @export
-#' @rdname validate_forecast
+#' @rdname assert_forecast
 #' @keywords check-forecasts
-validate_forecast.forecast_quantile <- function(forecast,
-                                                forecast_type = NULL, ...) {
-  forecast <- validate_general(forecast)
+assert_forecast.forecast_quantile <- function(
+  forecast, forecast_type = NULL, silent = FALSE, ...
+) {
+  forecast <- assert_forecast_generic(forecast)
   assert_forecast_type(forecast, actual = "quantile", desired = forecast_type)
   assert_numeric(forecast$quantile_level, lower = 0, upper = 1)
-  return(forecast[])
+  return(invisible(NULL))
 }
 
 
 #' @export
-#' @rdname validate_forecast
+#' @rdname assert_forecast
 #' @keywords check-forecasts
-validate_forecast.forecast_sample <- function(forecast, forecast_type = NULL, ...) {
-  forecast <- validate_general(forecast)
+assert_forecast.forecast_sample <- function(forecast, forecast_type = NULL, ...) {
+  forecast <- assert_forecast_generic(forecast)
   assert_forecast_type(forecast, actual = "sample", desired = forecast_type)
-  return(forecast[])
+  return(invisible(NULL))
+}
+
+
+#' @title Re-validate an existing forecast object
+#'
+#' @description
+#' The function re-validates an existing forecast object. It is similar to
+#' [assert_forecast()], but returns the input data instead of an invisible
+#' `NULL`. See [as_forecast()] for details on the expected input formats.
+#' @inherit assert_forecast params return examples
+#' @export
+#' @keywords check-forecasts
+validate_forecast <- function(forecast, forecast_type = NULL, silent = FALSE) {
```

Comment:
Is your namespace up to date? This isn't showing up is it?

Reply:
I think it's not showing up because there was previously a different `validate_forecast()` function